en/ros/ros2_comm.md

```diff
@@ -348,116 +428,147 @@ int main(int argc, char *argv[])
 }
 ```
 
-
 ### Offboard Control
 
 For a complete reference example on how to use Offboard control with PX4, see: [ROS 2 Offboard control example](../ros/ros2_offboard_control.md).
 
+## Using Flight Controller Hardware
 
-## Manual Workspace Setup with a ROS1 compatible workspace (FYI Only)
-
-:::note
-This is provided to help you better understand the build process and how to include the ROS1 workspace.
-It is not needed to build or use ROS 2.
-It additionally includes instructions for building the `ros1_bridge` package, which is used in [ROS (1) via ROS 2 Bridge](../ros/ros1_via_ros2.md).
-:::
+ROS2 with PX4 running on a flight controller is almost the same as working with PX4 on the simulator.
+The only difference is that you need to start both the agent _and the client_, with settings appropriate for the communication channel.
 
-This section describes the process to *manually* setup your workspace and build the `px4_ros_com`, `px4_msgs`, and `ros1_bridge` package.
-The topic effectively explains the operation of the `build_ros2_workspace.bash` script in the [installation instructions](#build-ros-2-workspace)).
+For more information see [Starting XRCE-DDS](../middleware/xrce_dds.md#starting-xrce-dds).
 
+## Custom uORB Topics
 
-**To build the ROS 2 workspace only:**
+ROS 2 needs to have the _same_ message definitions that were used to create the XRCE-DDS client module in the PX4 Firmware.
+These are automatically exported to [PX4/px4_msgs](https://github.com/PX4/px4_msgs) for main and release builds, and you can just clone them into your workspace.
 
-1. `cd` into `px4_ros_com_ros2` dir and source the ROS 2 environment.
-   Don't mind if it tells you that a previous workspace was set before:
-
-   ```sh
-   cd ~/px4_ros_com_ros2
-   source /opt/ros/foxy/setup.bash
-   ```
+If working on a custom build that has other messages, you would:
+- Update the [dds_topics.yaml](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/microdds_client/dds_topics.yaml) to add needed topics to the XRCE-DDS client in your PX4 firmware.
+- Copy the msg definitions out of the PX4 source tree into any ROS2 workspace that needs them.
+  The messages must be in a folder named `px4_msgs/msg/`.
+  
+  :::note
+  Technically the messages must be in a folder defined by the type prefix in the yaml file:
+  As you can see below, this is `px4_msgs::msg::`:
+  
+  ```yaml
+  - topic: /fmu/out/vehicle_odometry
+    type: px4_msgs::msg::VehicleOdometry
+  ```
+  :::
+  
+## ros2 CLI
```

Comment:
@beniaminopozzan FYI, I thought it useful to have the ros2 cli information here, where people might discover it. I think it is useful because PX4 users might not know about these, and they are a quick way of working out if topics are published (perhaps I should say that!)

ROS people might find them useful because they make clear you need the px4_msgs to do much other than list the topics.